systems/src/apu/mod.rs

```diff
@@ -0,0 +1,1396 @@
+                // with the master and start switches on.
+                assert_eq!(tester.start_contactor_energized(), false);
+
+                tester = tester.run(Duration::from_secs(1));
+                n = tester.get_n().get::<percent>();
+            }
+        }
+
+        #[test]
+        fn available_when_n_above_99_5_percent() {
+            let mut tester = tester_with().starting_apu();
+
+            loop {
+                tester = tester.run(Duration::from_millis(50));
+                let n = tester.get_n().get::<percent>();
+                assert!((n > 99.5 && tester.apu_is_available()) || !tester.apu_is_available());
+
+                if n == 100. {
+                    break;
+                }
+            }
+        }
+
+        #[test]
+        fn available_when_n_above_95_percent_for_2_seconds() {
+            let mut tester = tester_with()
+                .starting_apu()
+                .and()
+                .turbine_infinitely_running_at(Ratio::new::<percent>(96.))
+                .run(Duration::from_millis(1999));
+
+            assert!(!tester.apu_is_available());
+
+            tester = tester.run(Duration::from_millis(1));
+            assert!(tester.apu_is_available());
+        }
+
+        #[test]
+        #[timeout(500)]
+        fn without_fuel_apu_starts_until_approximately_n_3_percent_and_then_shuts_down_with_fault()
+        {
+            let mut tester = tester_with()
+                .no_fuel_available()
+                .run(Duration::from_secs(1_000))
+                .then_continue_with()
+                .starting_apu();
+
+            loop {
+                tester = tester.run(Duration::from_millis(50));
+                if tester.get_n().get::<percent>() >= 3. {
+                    break;
+                }
+            }
+
+            tester = tester.run(Duration::from_secs(10));
+
+            assert_eq!(tester.apu_is_available(), false);
+            assert_eq!(tester.has_fuel_low_pressure_fault(), true);
+            assert!(tester.master_has_fault());
+            assert!(!tester.start_is_on());
+        }
+
+        #[test]
+        fn starting_apu_shuts_down_when_no_more_fuel_available() {
+            let tester = tester_with()
+                .starting_apu()
+                .run(Duration::from_secs(10))
+                .then_continue_with()
+                .no_fuel_available()
+                .run(Duration::from_secs(1_000));
+
+            assert_eq!(tester.apu_is_available(), false);
+            assert_eq!(tester.has_fuel_low_pressure_fault(), true);
+            assert!(tester.master_has_fault());
+            assert!(!tester.start_is_on());
+        }
+
+        #[test]
+        fn running_apu_shuts_down_when_no_more_fuel_available() {
+            let tester = tester_with()
+                .running_apu()
+                .then_continue_with()
+                .no_fuel_available()
+                // Two runs, because of turbine state change from Running to Stopping.
```

Comment:
Ah, this explains the other case I commented on above. Could add a comment to the other one as well.